cpp/src/arrow/array.cc

```diff
@@ -48,6 +48,14 @@ bool Array::EqualsExact(const Array& other) const {
   return true;
 }
 
+bool Array::RangeEqualsExact(int32_t start_idx, int32_t end_idx, const Array& arr) const {
+  if (this == &arr) { return true; }
+  for (int i = start_idx; i < end_idx; ++i) {
```

Comment:
I think it's better to do some checks for `start_idx` and `end_idx`, to avoid always returning `true`.